### PR TITLE
Adds KAT tests for blake2s

### PR DESCRIPTION
**Pull request by suhithr**
Adds Known Answer Tests for BLAKE2S using test vectors from [the BLAKE2 repo](https://github.com/BLAKE2/BLAKE2)

Few small variable declarations were moved to the top of their respective functions to comply with the `-c89` standard flag

### CODE REVIEW

File: src/test.c [hunk 1]
```diff
@@ -8,51 +8,55 @@
...
-  printf("key   : %s\n", key);
   
-  clock_t start = clock();
-  for (i = 0; i <= buflen; ++i) {
-    blake2s(hash, BLAKE2S_OUTBYTES, buf, i, key, keylen);
+  for (i = 0; i < BLAKE2S_KEYBYTES; ++i)
+    key[i] = (uint8_t)i;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i)
+    buf[i] = (uint8_t)0;
```
Comment (franziskuskiefer) [inline src/test.c]: this could be done with a memset as well. But why is it necessary?

File: src/test.c [hunk 2]
```diff
@@ -8,51 +8,55 @@
...
-  size_t keylen = sizeof(key); 
   size_t i;
-
-  printf("input : %s\n", buf);
-  printf("key   : %s\n", key);
   
-  clock_t start = clock();
-  for (i = 0; i <= buflen; ++i) {
-    blake2s(hash, BLAKE2S_OUTBYTES, buf, i, key, keylen);
+  for (i = 0; i < BLAKE2S_KEYBYTES; ++i)
```
Comment (franziskuskiefer) [inline src/test.c:34]: please use braces everywhere

**Review by franziskuskiefer — COMMENTED** [review]
In addition to having warnings about missing new lines at the end of files this doesn't build for me (master neither though).

ld: can't open output file for writing: bin/blake2s, errno=2 for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

File: src/test.c (hunk 2 above)
Comment (suhithr) [reply src/test.c:34]: Will fix

File: src/test.c (hunk 1 above)
Comment (suhithr) [reply src/test.c]: I don't think it's necessary, I've removed it in an amended commit

**Comment by suhithr** [comment]
Regarding the building, I've made changes to the Makefile in 06f2353 it should work now

**Pull request merged**